src/file.cr

```diff
@@ -469,7 +469,8 @@ class File < IO::FileDescriptor
           byte_count -= 1
         end
 
-        str.write part.unsafe_byte_slice(byte_start, byte_count)
+        pointer = part.to_unsafe + byte_start
+        str.write Slice.new(pointer, byte_count, read_only: true)
```

Comment:
`str.write part.to_slice[pointer, byte_count]` should work...

Reply:
Ah, thanks, I’m still learning all of this. 🙂 

Unfortunately:
```
in src/file.cr:472: instantiating 'Slice(UInt8)#[](Pointer(UInt8), Int32)'

        str.write part.to_slice[part.to_unsafe + byte_start, byte_count]
                               ^

in src/slice.cr:195: no overload matches 'Int32#<=' with type Pointer(UInt8)
Overloads are:
 - Int32#<=(other : Int8)
 - Int32#<=(other : Int16)
 - Int32#<=(other : Int32)
 - Int32#<=(other : Int64)
 - Int32#<=(other : Int128)
 - Int32#<=(other : UInt8)
 - Int32#<=(other : UInt16)
 - Int32#<=(other : UInt32)
 - Int32#<=(other : UInt64)
 - Int32#<=(other : UInt128)
 - Int32#<=(other : Float32)
 - Int32#<=(other : Float64)
 - Comparable(T)#<=(other : T)

    unless 0 <= start <= @size
             ^
```

Reply:
Actually, should this be just the below?

```crystal
str.write part.to_slice[byte_start, byte_count]
```